### PR TITLE
feat: tests for activity component name and util refactor

### PR DESCRIPTION
**Pull request by tenbradley**
### Context:
[US124705](https://rally1.rallydev.com/#/357252966636d/dashboard?detail=%2Fuserstory%2F493234624660)
Creating tests for the name component and refactoring some functions to be used in general tests.

### Quality:
- 100% test coverage on d2l-activity-name-editor.js
- 100% test coverage on d2l-activity-name.js
- 100% test coverage on d2l-activity-name-course.js
- 100% test coverage on d2l-activity-name-learning-path.js

### CODE REVIEW

File: test/test-util.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// would prefer a better way
+export async function delayAndAwaitForElement(element, delayMs) {
+	return aTimeout(delayMs).then(async() => {
+		await elementUpdated(element);
+	});
+}
+
+// Handle updating the component with a new value and await the update
+export async function fireEventAndWait(element, eventType) {
+	const inputEvent = new Event(eventType);
```
Comment (AKobets) [inline test/test-util.js:22]: in initial version of this function  textarea was  dispatching event and await elementUpdated was done on the componentnt( parent of textarea). should this function have  three arguments? component, element, eventType ?

Comment (tenbradley) [reply test/test-util.js:22]: That seems reasonable.

Comment (tenbradley) [reply test/test-util.js:22]: I have just updated this function to accommodate waiting on a component instead of the element the event is being sent to.

**Comment by tenbradley** [comment]
Sorry. This pr ended up being larger than intended. Decided to combine the refactor of util function in to this pr to make life easier.

File: test/activity/description/d2l-activity-description-editor.test.js [hunk 2]
```diff
@@ -94,15 +77,15 @@ describe('d2l-activity-description-editor', () => {
...
+				assert.equal(element.shadowRoot.querySelector(textAreaLabel).value,
 					learningPathExisting.properties.description, 'textarea value does not match');
 
 				assert.equal(element.description, learningPathExisting.properties.description, 'description property should match');
 			});
 
 			it('updating should commit state', async() => {
 				const spy = sinon.spy(element.updateDescription);
-				await fireTextareaInputEvent(element);
+				await _fireTextareaInputEvent(element, updatedDescriptionText, updatedDescriptionText);
```
Comment (AKobets) [inline test/activity/description/d2l-activity-description-editor.test.js:88]: is it a typo in function call ? updatedDescriptionText, updatedDescriptionText`

File: test/activity/dialog/d2l-activity-dialog.test.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		beforeEach(async() => {
+			clearStore();
+		});
+
+		afterEach(() => {
+			mockLink.resetHistory();
+		});
+		it('should initialize using defined path and expected values', async() => {
+			await _createDialog('/learning-path/new');
+
```
Comment (AKobets) [inline test/activity/dialog/d2l-activity-dialog.test.js:32]: what else worth validating for this component in addition that the link was followed?? Does it have some properties that  are user facing?

File: test/activity/name/d2l-activity-name-editor.test.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+		describe('path:/learning-path/existing', () => {
+			let element;
+			beforeEach(async() => {
+				clearStore();
+				element = await _createNameEditor('/learning-path/existing');
+				assert.equal(element.name, learningPathExisting.properties.name, 'description should match response');
+			});
+
+			it('name should be set when one is provided', async() => {
+				// new path should be followed
```
Comment (AKobets) [inline test/activity/name/d2l-activity-name-editor.test.js:61]: the test description does not match assertions

File: test/activity/name/d2l-activity-name-editor.test.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+			assert.equal(element.name, learningPathNew.properties.name);
+		});
+
+		describe('path:/learning-path/existing', () => {
+			let element;
+			beforeEach(async() => {
+				clearStore();
+				element = await _createNameEditor('/learning-path/existing');
+				assert.equal(element.name, learningPathExisting.properties.name, 'description should match response');
```
Comment (AKobets) [inline test/activity/name/d2l-activity-name-editor.test.js:57]: should be part of some test ?

File: test/activity/name/d2l-activity-name-editor.test.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+			it('extra whitespace is removed', async() => {
+				const spy = sinon.spy(element.updateName);
+				await updateName(element, '   new  name     ');
+
+				assert.equal(element.name, 'new  name', 'name was updated to match');
+				assert.isTrue(spy.commit.called, 'commit should be called when input event is triggered');
+			});
+
+			it('empty name is not committed', async() => {
+				const spy = sinon.spy(element.updateName);
```
Comment (AKobets) [inline test/activity/name/d2l-activity-name-editor.test.js:83]: test should get name before calling update, the update with empty value and then assert that commit not called and name was not changed. It should not compare  against learningPathExisting.properties.name

File: test/activity/name/d2l-activity-name-editor.test.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+		it('should construct', () => {
+			runConstructor('d2l-activity-name-editor');
+		});
+	});
+
+	describe('Component', () => {
+
+		beforeEach(async() => {
+			clearStore();
```
Comment (AKobets) [inline test/activity/name/d2l-activity-name-editor.test.js:35]: why it is async without await inside ?

File: test/activity/name/d2l-activity-name-editor.test.js (hunk 5 above)
Comment (AKobets) [reply test/activity/name/d2l-activity-name-editor.test.js:57]: `description` should match response ?

File: test/activity/name/d2l-activity-name.test.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+	return await createComponentAndWait(html`<d2l-activity-name-course href="${path}" token="test-token"></d2l-activity-name-course>`);
+}
+
+async function updateName(inputArea, updatedText, component) {
+	inputArea.value = updatedText;
+
+	await fireEventAndWait(inputArea, 'input', component);
+}
+
+describe('d2l-activity-description', () => {
```
Comment (AKobets) [inline test/activity/name/d2l-activity-name.test.js:25]: typo `d2l-activity-description`

File: test/activity/name/d2l-activity-name.test.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+	describe('constructor', () => {
+		it('should construct d2l-activity-name', () => {
+			runConstructor('d2l-activity-name');
+		});
+
+		it('should construct d2l-activity-name-learning-path', () => {
```
Comment (AKobets) [inline test/activity/name/d2l-activity-name.test.js:32]: two tests with same name/body

File: test/activity/name/d2l-activity-name.test.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+	describe('activity-name-course', () => {
+
+		it('creation', async() => {
+			const element = await _createNameCourse('/course/existing');
+			assert.instanceOf(element, ActivityNameCourse, 'course name component should exist');
+		});
+	});
+	it('create course from learning path', async() => {
+		const element = await fixture(html`<d2l-activity-name></d2l-activity-name>`);
```
Comment (AKobets) [inline test/activity/name/d2l-activity-name.test.js:86]: what does it mean `create course from learning path`

File: test/activity/name/d2l-activity-name.test.js [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+		beforeEach(async() => {
+			clearStore();
+			element = await _createNameLearningPath('/learning-path/existing');
+			assert.equal(element.name, learningPathExisting.properties.name, 'name should match response');
+		});
+
+		it('creation', async() => {
+			element = await _createNameLearningPath('/learning-path/new');
+			assert.equal(element.name, learningPathNew.properties.name, 'name should be blank in new learning path');
+		});
```
Comment (lloydwaddell) [inline test/activity/name/d2l-activity-name.test.js:50]: this doesn't benefit from the beforeEach, should be its own describe block

**Review by lloydwaddell — APPROVED** [review]

**Pull request merged**